code/3_data_processing.py

```diff
@@ -45,67 +45,89 @@
 
 import os
 import sys
+
     
 def main():
-  
-    from pyspark.sql import SparkSession
-    from pyspark.sql import functions as sqlfn
-
-    spark = (
-        SparkSession.builder.appName("Airline Data Exploration")
-        .config("spark.executor.memory", "8g")
-        .config("spark.executor.cores", "4")
-        .config("spark.driver.memory", "20g")
-        .config("spark.executor.instances", "4")
-        .config("spark.yarn.access.hadoopFileSystems", os.environ["STORAGE"])
-        .getOrCreate()
-    )
+    import cml.data_v1 as cmldata
+    from pyspark import SparkContext
+    from pyspark.sql import functions as sqlfn   
+    
+    ### cleanup needed start
+    # These should come from the AMP
+    os.environ['SPARK_CONNECTION_NAME'] = "go01-aw-dl"
+    os.environ['DW_DATABASE'] = "airlines_iceberg"
+    os.environ['DW_TABLE'] = "flights"
```

Comment:
This is a good one. Unfortunately, we don't have a default spark connection, it's always named after the data lake. I added this exception handling, it was a bit more complicated because it's not enough to update the env var within the session I needed to change it in the project too - good APIv2 example landed. 

I run 3 test: local, external, external fallback to local, all completed with success.